### PR TITLE
Migrating clib-package and clib-cache to the main repository.

### PR DESCRIPTION
**Pull request by Isty001**
As #145 says, it's pretty painful to keep this code in different repositories. 

This PR migrates clib-package and clib-cache repositories to the main repo, including tests.
Also a new option -c / --sikp-cache is available for `clib install`, and moved the search cache to the other caching functions. 

Please review if everything is ok.

### CODE REVIEW

**Comment by stephenmathieson** [comment]
This PR is pretty huge, so it's rather difficult to review. I'll try to take an in-depth look tomorrow.

File: src/bin/clib-search.c [hunk 1]
```diff
@@ -151,6 +114,9 @@ main(int argc, char *argv[]) {
 
   debug_init(&debugger, "clib-search");
 
+  // 15 days
+  clib_cache_init(15 * 24 *60 * 60);
```
Comment (stephenmathieson) [inline src/bin/clib-search.c]: Not sure what's happening here, but 15 days seems pretty arbitrary. I'm thinking we should make this configurable.

File: test/cache-test.c [hunk 2]
```diff
@@ -0,0 +1,149 @@
+#include <limits.h>
+#include <string.h>
+#include "rimraf/rimraf.h"
+#include "minunit/minunit.h"
```
Comment (stephenmathieson) [inline test/cache-test.c]: I'm -1 on having two unit testing frameworks here. If we're going to move everything here, let's standardize on a few things:

- code style
- tests/test frameworks
- build tools
- ... etc

**Comment by Isty001** [comment]
Okay, sure. I'll take another look at it soon, and update this PR.

**Pull request closed without merging**